api/v1alpha2/kubeadmbootstrapconfig_types.go

```diff
@@ -61,3 +64,18 @@ type KubeadmConfigList struct {
 func init() {
 	SchemeBuilder.Register(&KubeadmConfig{}, &KubeadmConfigList{})
 }
+
+// Files defines the input for generating write_files in cloud-init.
+type Files struct {
+	// Path specifies the full path on disk where to store the file.
+	Path string `json:"path"`
+
+	// Owner specifies the ownership of the file, e.g. "root:root".
+	Owner string `json:"owner"`
+
+	// Permissions specifies the permissions to assign to the file, e.g. "0640".
+	Permissions string `json:"permissions"`
```

Comment:
Permissions should probably be optional, since it isn't required for the cloud-init module.

Reply:
Fixed, PTAL

Reply:
The data type lgtm, but I suspect that the template will need to be updated here: https://github.com/kubernetes-sigs/cluster-api-bootstrap-provider-kubeadm/blob/64d6676d48aab6ae55c36435176f147e03ea8a7a/cloudinit/files.go#L24-L25 as well.

Reply:
Good point. Working on fixing that now.